src/Sarif.Converters.UnitTests/PyLintConverterTests.cs

```diff
@@ -0,0 +1,164 @@
+            {
+                Uri = new Uri("test.py", UriKind.RelativeOrAbsolute),
+                Region = region
+            };
+
+            Location location = new Location
+            {
+                AnalysisTarget = physloc
+            };
+
+            testResult.Locations = new List<Location>()
+            {
+                location
+            };
+
+            return testResult;
+        }
+
+        [Fact]
+        public void PyLintConverter_Convert_WhenInputIsNull_ThrowsArgumentNullException()
+        {            
+
+            var converter = new PyLintConverter();
+            var mockLogWriter = new Mock<IResultLogWriter>();
+
+            Action action = () => converter.Convert(null, mockLogWriter.Object, Microsoft.CodeAnalysis.Sarif.Writers.LoggingOptions.None);
+            action.ShouldThrow<ArgumentNullException>();            
+        }
+
+        [Fact]
+        public void PyLintConverter_Convert_WhenInputIsValid_Passes()
+        {
+            byte[] data = Encoding.UTF8.GetBytes(CreateTestPyLintStringLogList());
+            MemoryStream stream = new MemoryStream(data);
+
+            var mockWriter = new Mock<IResultLogWriter>();
+            mockWriter.Setup(writer => writer.Initialize(It.IsAny<string>(), It.IsAny<string>()));
+            mockWriter.Setup(writer => writer.WriteTool(It.IsAny<Tool>()));
+            mockWriter.Setup(writer => writer.WriteFiles(It.IsAny<IDictionary<string, FileData>>()));
+            mockWriter.Setup(writer => writer.OpenResults());
+            mockWriter.Setup(writer => writer.CloseResults());
+            mockWriter.Setup(writer => writer.WriteResults(It.IsAny<List<Result>>()));
+
+            var converter = new PyLintConverter();
+            
+            converter.Convert(stream, mockWriter.Object, Microsoft.CodeAnalysis.Sarif.Writers.LoggingOptions.None);
+                        
+            mockWriter.Verify(writer => writer.Initialize(It.IsAny<string>(), It.IsAny<string>()), Times.Once);
+            mockWriter.Verify(writer => writer.WriteTool(It.IsAny<Tool>()), Times.Once);
+            mockWriter.Verify(writer => writer.WriteFiles(It.IsAny<IDictionary<string, FileData>>()), Times.Once);
+            mockWriter.Verify(writer => writer.OpenResults(), Times.Once);
+            mockWriter.Verify(writer => writer.CloseResults(), Times.Once);
+            mockWriter.Verify(writer => writer.WriteResults(It.IsAny<List<Result>>()), Times.Once);
+        }
+        
+        [Fact]
+        public void PyLintConverter_Convert_WhenOutputIsNull_ThrowsArgumentNullException()
+        {
+            var converter = new PyLintConverter();
+
+            Action action = () => converter.Convert(new MemoryStream(), null, Microsoft.CodeAnalysis.Sarif.Writers.LoggingOptions.None);
+            action.ShouldThrow<ArgumentNullException>();
+        }
+
+        [Fact]
+        public void PyLintConverter_PrepareSarifResult_WhenInputIsNull_ThrowsArgumentNullException()
```

Comment:
Per my suggested rename below: In these two test methods, change `PrepareSarifResult` to `CreateResult`.